backend/src/facebook-feed/facebook-feed.service.ts

```diff
@@ -19,34 +21,40 @@ export class FacebookFeedService {
     }
 
     async fetchFacebookFeeds() {
-        try {
-            this.logger.log('Fetching facebook feeds Started');
-            const vantaaFacebookDataUrl = `${config.facebookGraphApi}${sources.vantaa}/posts?`
-                + `fields=${queryParams.join(',')}`
-                + `&access_token=${config.facebookPageToken}`;
-            const facebookFeeds = this.transformData((await axios.get(vantaaFacebookDataUrl)).data.data);
-            await this.feedService.saveFeeds(facebookFeeds);
-            this.logger.log('Fetching facebook feeds Completed');
-        } catch (error) {
-            this.logger.error(`Failed to get facebook feeds: ${error}`)
-        }
+        this.logger.log('Fetching facebook feeds Started');
+        FB.api(`${sources.vantaa}/posts`, { fields: queryParams })
+            .then(this.transformData)
+            .then(this.persistIntoDb)
+            .catch(error => this.logger.error(`Failed to get facebook feeds: ${error}`));
+        this.logger.log('Fetching facebook feeds Completed');
+    }
+
+    persistIntoDb = feeds => this.feedService.saveFeeds(feeds);
+
+    transformData = async (feeds) => {
```

Comment:
`transformData` should be pure function so that you can easily test it.
No remote data fetching should be put here.